### DISCUSSION TITLE
Problem running ADT on M2 Mac

### DISCUSSION THREAD

**Discussion by fh755** [airsdk/Adobe-Runtime-Support#2639] [General]
Hi all, I've recently moved from PC to Mac and have followed the instructions at airsdk.dev to set everything up. I used the AIR SDK manager which worked well and installed AIR 50.2.2.5. I also installed Android Studio and JDK 11. And I removed the quarantine from the AIR SDK.

But I'm having issues running the ADT command, which worked fine on my PC.

My command is:
cd ~/Documents/TestCompile     
(this folder contains all files - Test-app.xml Test.swf icon36.png icon48.png icon72.png icon96.png icon144.png icon192.png icon512.png music.mp3)

then:
/Users/XXXXX/sdks/air/AIRSDK_50.2.2/bin/adt -package -target apk-captive-runtime -storetype pkcs12 -keystore XXXXX.p12 -storepass XXXXXX Test.apk Test-app.xml Test.swf icon36.png icon48.png icon72.png icon96.png icon144.png icon192.png icon512.png music.mp3

The AIR SDK Developer Tool Log from the AIR SDK Manager shows:
Target requested was apk-captive-runtime, but not using legacy APK so building via Android Studio
UsageError: no such file /private/var/folders/6c/952krrhd4nz16kvbcqbpddyr0000gn/T/hsperfdata_XXXXX/Test.swf

I've searched the forums but I can't find anything similar. Appreciate any advice, thanks

**Comment by ajwfrost** [comment]
That's an odd error to have when packaging an APK. It's got the Test.swf file at a temporary file location .. but when we're packaging using Android Studio / Gradle, we create a project structure and copy the Test.swf into this (which means, even if the Android Studio project is under a temp location, the path to Test.swf will include app/src/main/assets/Test.swf)

Is there any call stack available for that UsageError? I'm wondering whether there's some set-up where it's trying to run the whole application within a sandboxed location i.e. the initial ADT runs but then it re-loads itself using a sandbox, but without copying over the relevant files. I've not seen this kind of result though...

My suggestion might be to run the command again but this time generate the Android Studio project itself rather than trying to do the APK in one step. This would at least help narrow down where the issue might be ... so if you use the same command line, just change `apk-captive-runtime` to `android-studio`, and `Test.apk` to `AndroidStudioProject` or similar, and see if that completes successfully. If it does, you can open the `AndroidStudioProject` folder within Android Studio and build the APK/bundle from there....

thanks

**Reply by fh755** [reply]
Thanks for your quick reply Andrew. I tried what you suggested with generating the Android Studio project itself but I got the exact same error message.

But I've got it working now. I just moved the folder where my files were out of the Documents folder. 

So from:
cd ~/Documents/TestCompile
to 
cd /Users/myname/TestCompile

All is working now. I must say I'm impressed with the work you and HARMAN are doing with the AIR, thanks!

**Reply by ajwfrost** [reply]
Ah, so that sounds like it's just some weird thing related to Apple's filesystem security..  they are good at making it safe and simple for general purpose users, but it does make it harder for developers I think!
Thanks :-)
